### PR TITLE
add acceptance tests for all help commands

### PR DESCRIPTION
**Pull request by kellyselden**
Also removed duplicate aliases, as well as reordered aliases for consistency.

Adding upon #4577 to help prevent regressions with #4576.

### CODE REVIEW

**Comment by stefanpenner** [comment]
i would love to see future related PR, moving away from runCommand, to instead invoke programmatically ala ember helper.

**Comment by kellyselden** [comment]
@stefanpenner I agree 100%. I noticed how much I bloated the run time of CI.

**Comment by stefanpenner** [comment]
Maybe this pr should wait until it uses the direct api(that stuff may need some tweaks). I nervous to slow down the tests even more

**Comment by kellyselden** [comment]
Ok sounds good

**Comment by kellyselden** [comment]
@stefanpenner My first test rewrite went from (1208ms) to (291ms) (Windows of course...). Thanks for the suggestion.

**Comment by kellyselden** [comment]
@stefanpenner All changes made. My tests pass locally. Looks like the fingerprinting issue is back on CI?

**Comment by kellyselden** [comment]
@stefanpenner Do you have any suggestions on how to move this forward? I can't get the CI to go green, but I think the failures are irrelevant. All these `help` tests pass locally.

**Comment by stefanpenner** [comment]
i'll take a look once they fail again, and see if i can provide some help.

**Comment by stefanpenner** [comment]
@rickharrison any thoughts on what the md5 is differing between CI and local all of a sudden?

**Comment by kellyselden** [comment]
@rickharrison It's weird. I keep rebasing from master that had this fixed a while ago, yet master is green and I can't seem to eliminate it from this branch/PR.

**Comment by stefanpenner** [comment]
i think i found it: https://github.com/rickharrison/broccoli-asset-rev/blob/master/lib/fingerprint.js#L151

**Comment by rickharrison** [comment]
Hmmm the whole point of the recent fix in asset rev was to prevent differences in hashing. I'll be around in about an hour if you want to chat about it on irc.

**Comment by stefanpenner** [comment]
> Hmmm the whole point of the recent fix in asset rev was to prevent differences in hashing. I'll be around in about an hour if you want to chat about it on irc.

https://github.com/rickharrison/broccoli-asset-rev/pull/69

**Comment by stefanpenner** [comment]
@kellyselden my above PR may sort out the issue.

**Comment by rickharrison** [comment]
Published to npm as 2.1.2

**Comment by stefanpenner** [comment]
@kellyselden i suspect the key may now need to be updatd.

**Comment by kellyselden** [comment]
@stefanpenner Can you explain what you mean by key? The test generates the expected key using the crypto library, so I don't think it's a hardcoded hash I have to change. The test passes locally for me. So strange.

**Comment by stefanpenner** [comment]
> So strange.

ya funky

**Comment by stefanpenner** [comment]
I'm very eager to get this in, those runCommand -> ember calls look nice time wise (and likely debuggability) 

Im going to poke around and see if i can figure out whats up.

File: tests/acceptance/smoke-test-slow.js [hunk 1]
```diff
@@ -81,7 +81,11 @@ describe('Acceptance: smoke-test', function() {
       });
   });
 
-  it('ember new foo, build production and verify fingerprint', function() {
+  // there is a bug in here when running the entire suite on Travis
+  // when run in isolation, it passes
+  // here is the error:
+  // test-support-80f2fe63fae0c44478fe0f8af73200a7.js contains the fingerprint (2871106928f813936fdd64f4d16005ac): expected 'test-support-80f2fe63fae0c44478fe0f8af73200a7.js' to include '2871106928f813936fdd64f4d16005ac'
+  it.skip('ember new foo, build production and verify fingerprint', function() {
```
Comment (kellyselden) [inline tests/acceptance/smoke-test-slow.js]: just pointing this out.

**Comment by kellyselden** [comment]
OMG green travis! @stefanpenner What do you think? I'm not sure why appveyor failed.

**Comment by stefanpenner** [comment]
@kellyselden i don't believe your color-code specific tests work on windows.

@trabus any thoughts, since your our ASCII color dude?

**Comment by kellyselden** [comment]
@stefanpenner huh. I developed this on Windows, and the tests were passing for me. Maybe it was the shell I was using. I could always insert the chalk calls in my tests.

**Comment by stefanpenner** [comment]
> @stefanpenner huh. I developed this on Windows, and the tests were passing for me. Maybe it was the shell I was using. I could always insert the chalk calls in my tests.

windows is :(

**Comment by kellyselden** [comment]
For reference to help us solve this:
capture

**Comment by trabus** [comment]
@kellyselden The Windows default cmd.exe [does not support ANSI](http://superuser.com/questions/249128/coloring-windows-xp-cmd-exe) out of the box. For your tests to pass, I think you'll need to find a way to assert against the text sans-ANSI codes. :\

**Comment by kellyselden** [comment]
@trabus I wonder why it is passing on my cmd.exe. Maybe something I have installed is handling the color codes?

**Comment by trabus** [comment]
@kellyselden It's almost certain, since you've got color in your output. There's [hundreds of articles on how to enable it](https://www.google.com/search?q=adding+ascii+to+windows+cmd&oq=adding+ascii+to+windows+cmd), it may be that some tool you installed did it for you.

**Comment by kellyselden** [comment]
@trabus Thanks. I'll try disabling it and seeing where that gets me.

**Comment by trabus** [comment]
Our other option may be to see if we can get Appveyor to hook up the ANSI.SYS on our runner. When the codes are not present, they're simply ignored, and do not affect the function of ember-cli. It's just color formatting on console output, nothing critical.

@stefanpenner @raytiley  ^^ thoughts?

**Comment by kellyselden** [comment]
I got the windows CI to pass. I ended up replicating the chalk fallback in an extremely simple sense. Basically:
```js
  if (!supportsColor) {
    return stripAnsi(helpString);
  }
```
I tried enabling color in appveyor using [this as help](http://help.appveyor.com/discussions/suggestions/197-support-ansi-color-codes), but I couldn't get it working.
Is this an acceptable solution? @trabus @stefanpenner @raytiley

**Comment by stefanpenner** [comment]
@kellyselden LVGTM

**Comment by kellyselden** [comment]
Nice! Thanks.

**Pull request merged**